### PR TITLE
DG: user stories, manual testing and design considerations

### PR DESCRIPTION
**Pull request by ong-yinggao98**
Also updated PPP

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -466,6 +466,16 @@ The following activity diagram summarises the flow of events when `DeleteQuestio
...
 
+##### 6.3.1.4 Design considerations
+
+The `Student` class guarantees immutability, including its list of `Question` objects. As such, any operation must return a modified copy instead of directly modifying the base list.
+
+The pros of doing so is that it becomes easier to test and debug as there are fewer mutations to worry about.
+
+The downside, however, that this approach is more memory-intensive and possibly slower, since changing a single question involves recreating the entire list of questions attached to the student.
+
+Having considered that our target audience (one-to-one private tutors) are unlikely to have so much data that this would severely impact performance, we believe that this is worth the trade-off.
```
Comment (AlexCQY) [inline docs/DeveloperGuide.md:477]: Does this contradict our NFR as one of them is "The program should be able to handle at least 100 students."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1042,8 +1055,103 @@ testers are expected to do more *exploratory* testing.
...
+    1. Prerequisite: List all students using `list` command. View academic data with `toggle` command. First student on the list must have at least one question.
+
+    1. Test case: `question solve 0 i/1`
+       Expected: No question deleted. Error details shown in status message.
+
+    1. Test case: `question solve 1 i/0`
+       Expected: Similar to previous.
+
+    1. Test case: `question solve 1 i/QUESTION_INDEX`
+       Expected: Question at `QUESTION_INDEX` is removed. Details of deleted question included in status message.
```
Comment (AlexCQY) [inline docs/DeveloperGuide.md]: Should this be "delete'?

Comment (ong-yinggao98) [reply docs/DeveloperGuide.md]: Thanks for the catch!

**Review by AlexCQY — APPROVED** [review]
lgtm

**Pull request merged**